pages/client-libraries/java.mdx

```diff
@@ -4,105 +4,687 @@ description: Peruse our guide on implementing Java in Memgraph. Comprehensive do
+Java AutoCloseable interface to manage the lifecycle of the client. The lifecycle of the client is tied to your application lifecycle.
+
+The following implementation of `AutoCloseable` interface will make sure to close the client connection once the application is finished: 
+
+```java
+@Override
+public void close() throws RuntimeException {
+    driver.close();
+}
 ```
-Node 1: Hello, World!
+
+Keep in mind that driver object is thread safe, and can be reused between different threads. 
+
+### Query the database 
+
+After connecting your driver to the Memgraph you can start running some queries, the simplest way to run the query is via writeTransaction and readTransaction methods.
+
+#### Run a write query 
+
+The following query will create a node inside the database: 
+
+```java
+public void createNode() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        session.writeTransaction(tx -> {
+            tx.run("CREATE (n:Technology {name:'Memgraph'})");
+            return null;
+        });
+    }
+}
 ```
 
-<Callout type="info">
-Memgraph created [Bolt Java Driver](https://github.com/memgraph/bolt-java-driver) which can be used to connect to a running Memgraph instance. We still recommend you use the above mentioned driver.
-</Callout>
\ No newline at end of file
+The session object takes a SessionConfig object as its argument. The SessionConfig object is used to specify the database name, and other session configuration options. Session API supports writeTransaction and readTransaction methods.
+The writeTransaction method of a Neo4j Session object takes a TransactionWork functional interface as its argument. The TransactionWork interface has a single method execute, which takes a Transaction object as its argument and returns a value of type T, void in this case. 
+
+#### Run a read query 
+
+Similar to the write query, there is a `readTranasction()` method that can be used to run read queries. The difference is that read will actually return the results of the query.
+
+The following query will read data from the database: 
+
+```java
+public void readNode() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        List<Record> result = session.readTransaction(tx -> {
+            Result queryResult = tx.run("MATCH (n:Technology{name: 'Memgraph'}) RETURN n");
+            List<Record> records = queryResult.list();
+            return records;
+        });
+        System.out.println(result);
+
+        // Print each node as map
+        for (Record record : result) {
+            System.out.println(record.get("n").asMap());
+        }
+    }
+}
+```
+
+In this query, each `Record` contains a `Node` accessible by the `asMap()` method.
+
+#### Run a query with parameters 
```

Comment:
In Go, this section i called Running queries with property map

Reply:
```suggestion
#### Running queries with property map
```